EIPS/eip-7748.md

```diff
@@ -0,0 +1,292 @@
+        state._conversion_curr_account.phase = StoragePhase("0x00")
+
+# Modified function: add new only_if_empty optional parameter.
+def set_storage(
+    state: State, addr: Address, key: Bytes, value: U256, only_if_empty: bool = True
+) -> None:
+    # <new_code>
+    if only_if_empty:
+        value = state._overlay_tree.get(get_tree_key_for_storage_slot(addr, key))
+        if value is not None:
+            return
+    # </new_code>
+    
+    state._overlay_tree.set(get_tree_key_for_storage_slot(addr, key), value)
+```
+
+As mentioned previously, the next function is called by `apply_body(...)` to perform the conversion step for a block:
+
+```python
+# Note the following function is optimized for readability, not for performance.
+def state_convert(state: State, stride: int):
+    n = 0    
+    while n < stride and not state._conversion_finished:
+        curr_account = state.get_conversion_account()
+        
+        # EIP-161 should not be converted.
+        if account_exists_and_is_empty(state, curr_account.address):
+            state.conversion_move_to_next_account()
+            continue
+        
+        # Account storage.
+        if curr_account.phase is StoragePhase:
+            # Get the storage-slot from _storage_tries which is MPT data.
+            trie = state._storage_tries.get(curr_account.address)
+            
+            if trie is not None:
+                slot_num, slot_value, next_key = trie_get_next_at_key(trie, curr_account.phase.next_key)
+                # The Overlay Tree will write in the VKT. We use the new
+                # only_if_empty parameter to avoid writing stale values.
+                set_storage(state, curr_account.address, slot_num, slot_value, only_if_empty=True)
+                n += 1
+    
+                if next_key is not None:
+                    # There're more storage-slots to be converted, continue in this phase.
+                    state.conversion_curr_account.phase.next_key = next_key
+                else:
+                    # No more storage-slots. Move to the code migration starting
+                    # at chunk-number zero.
+                    state.conversion_curr_account.phase = CodePhase(0)
+            else:
+                # There's no storage trie for the account, move directly to
+                # migrating code (if any).
+                state.conversion_curr_account.phase = CodePhase(0)
+        # Account code and basic data.
+        else:
+            # Getting the code from the Overlay Tree is fine since promises returning
+            # the Account full code which would come from the MPT or a separate code database.
+            account = get_account(state, curr_account.address)
+            chunked_code = chunkify_code(account.code)
+            
+            for chunk_num in range(len(chunked_code)):
+                state_set_codechunk(state, address, chunk_num, chunked_code[chunk_num])
+                n += 1
```

Comment:
ok so its possible for n > stride  depending on code

Reply:
Correct, this clarified in the `Rationale/Account code chunking done in a single step` subsection since might not be obvious why was decided this way.